From: Tom Herbert <tom@quantonium.net>
Date: Thu, 28 Dec 2017 11:00:44 -0800
Subject: strparser: Call sock_owned_by_user_nocheck
Patch-mainline: v4.15-rc6
Git-commit: d66fa9ec53c43bba9fa973c16419f6061b7cc3ea
References: bsc#1076830

strparser wants to check socket ownership without producing any
warnings. As indicated by the comment in the code, it is permissible
for owned_by_user to return true.

Fixes: 43a0c6751a322847 ("strparser: Stream parser for messages")
Reported-by: syzbot <syzkaller@googlegroups.com>
Reported-and-tested-by: <syzbot+c91c53af67f9ebe599a337d2e70950366153b295@syzkaller.appspotmail.com>
Signed-off-by: Tom Herbert <tom@quantonium.net>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Michal Kubecek <mkubecek@suse.cz>

---
 net/strparser/strparser.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/strparser/strparser.c b/net/strparser/strparser.c
index b5c279b22680..03475d633e19 100644
--- a/net/strparser/strparser.c
+++ b/net/strparser/strparser.c
@@ -365,7 +365,7 @@ void strp_data_ready(struct strparser *strp)
 	 * allows a thread in BH context to safely check if the process
 	 * lock is held. In this case, if the lock is held, queue work.
 	 */
-	if (sock_owned_by_user(strp->sk)) {
+	if (sock_owned_by_user_nocheck(strp->sk)) {
 		queue_work(strp_wq, &strp->rx_work);
 		return;
 	}
-- 
2.16.2

